manager/volume.go

```diff
@@ -614,7 +617,59 @@ func (m *VolumeManager) Expand(volumeName string, size int64) (v *longhorn.Volum
 	}
 
 	if v.Spec.Size > size {
-		return nil, fmt.Errorf("cannot expand volume %v with current size %v to a smaller size %v", v.Name, v.Spec.Size, size)
+		return nil, fmt.Errorf("longhorn volume %v spec size %v is greater than requested Persistent Volume Claim size %v", v.Name, v.Spec.Size, size)
+
+	}
+
+	if v.Status.ActualSize > size {
+		return nil, fmt.Errorf("longhorn volume %v actual size %v is greater than requested Persistent Volume Claim size %v", v.Name, v.Spec.Size, size)
+	}
+
+	kubernetesStatus := &v.Status.KubernetesStatus
+	if kubernetesStatus.PVCName != "" && kubernetesStatus.LastPVCRefAt == "" {
+		pvc, err := m.ds.GetPersistentVolumeClaim(kubernetesStatus.Namespace, kubernetesStatus.PVCName)
+		if err != nil {
+			return nil, err
+		}
+
+		requestedSize := resource.MustParse(strconv.FormatInt(size, 10))
+
+		// TODO: Should check for pvc.Spec.Resources.Requests.Storage() here, once upgrade API to v0.18.x.
+		storageSpecValue, ok := pvc.Spec.Resources.Requests[corev1.ResourceStorage]
+		if !ok {
+			return nil, fmt.Errorf("cannot get request storage")
+		}
+		// TODO: Should check for pvc.Status.CapacityStorage() here, once upgrade API to v0.18.x.
+		storageStatusValue, ok := pvc.Status.Capacity[corev1.ResourceStorage]
```

Comment:
Why we need to check status value?

Reply:
I remember during testing PVC status storage size didn't change till PV storage size is changed.  When failed, PVC storage request has updated value and PVC storage status has older value.  Use it to check for corner cases.

Reply:
As I said in https://github.com/longhorn/longhorn-manager/pull/888#discussion_r643003422, but I am fine with keeping it.